src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/SolutionExplorerInProcess.cs

```diff
@@ -682,4 +704,45 @@ private async Task<EnvDTE.Project> GetProjectAsync(string nameOrFileName, Cancel
                 });
         }
     }
+
+    internal sealed class UpdateSolutionEvents : IVsUpdateSolutionEvents, IDisposable
+    {
+        private uint _cookie;
+        private readonly IVsSolutionBuildManager2 _solutionBuildManager;
+
+        public event Action? OnUpdateSolutionDone;
+
+        internal UpdateSolutionEvents(IVsSolutionBuildManager2 solutionBuildManager)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            _solutionBuildManager = solutionBuildManager;
+            ErrorHandler.ThrowOnFailure(solutionBuildManager.AdviseUpdateSolutionEvents(this, out _cookie));
+        }
+
+        int IVsUpdateSolutionEvents.UpdateSolution_Begin(ref int pfCancelUpdate) => 0;
```

Comment:
I'm not sure if it's honored here, but much of VS allows you to return E_NOTIMPL as a way to say "I don't care about this method" and you get removed for future calls.

Reply:
I'll update them.